File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        res = 0
        count = 0
        def dfs( n , stair):
            nonlocal count , res

            count += stair

            if count == n:
                res +=1
                count -= stair
                return
            
            if count > n:
                count -= stair
                return

            

            dfs(n , 1)
            dfs(n , 2)

            count -= stair
        dfs(n , 0)
        return res

File: Data Structures & Algorithms/permutations/submission-1.cpp
class Solution {
public:
    vector<vector<int>> res;
    
    void dfs(vector<int>& nums ,vector<int> temp ,int num , vector<int> cur){
        if(cur.size() == nums.size()){
            res.push_back(cur);
            return;
        }
        erase(temp , num);

        for(int i= 0 ;i < temp.size(); i++){
            cur.push_back(temp[i]);
            
            dfs( nums ,temp , temp[i] , cur);
            cur.pop_back();
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        vector<int> cur;
        vector<int> temp = nums;
        dfs(nums ,temp ,INT_MIN , cur);
        return res;
    }
};
